features/dd-sdk-android-flags/src/test/kotlin/com/datadog/android/flags/featureflags/FlagsClientTest.kt

```diff
@@ -337,19 +337,77 @@ internal class FlagsClientTest {
         assertThat(client2).isInstanceOf(NoOpFlagsClient::class.java)
     }
 
+    @Test
+    fun `M throw IllegalArgumentException W Builder() {blank name}`() {
+        // When/Then
+        org.junit.jupiter.api.assertThrows<IllegalArgumentException> {
+            FlagsClient.Builder("", mockSdkCore)
+        }
+    }
+
+    @Test
+    fun `M throw IllegalArgumentException W Builder() {whitespace-only name}`() {
+        // When/Then
+        org.junit.jupiter.api.assertThrows<IllegalArgumentException> {
+            FlagsClient.Builder("   ", mockSdkCore)
+        }
+    }
+
     @Test
     fun `M return NoOpClient W Builder#build() {flags feature not enabled}`() {
+        // Given - create a mock SDK core with no feature enabled
+        val mockUninitializedCore = mock<FeatureSdkCore>()
+        val mockLogger = mock<InternalLogger>()
+        whenever(mockUninitializedCore.internalLogger).thenReturn(mockLogger)
+        whenever(mockUninitializedCore.getFeature(FLAGS_FEATURE_NAME)).thenReturn(null)
+
         // When - build when feature is not enabled
-        val client = FlagsClient.Builder(sdkCore = mockSdkCore).build()
+        val client = FlagsClient.Builder(sdkCore = mockUninitializedCore).build()
 
         // Then - should return NoOpClient
         assertThat(client).isInstanceOf(NoOpFlagsClient::class.java)
+
+        // Verify error was logged
+        verify(mockLogger).log(
+            eq(InternalLogger.Level.ERROR),
+            eq(InternalLogger.Target.USER),
+            any(),
```

Comment:
It is better to verify the message as well, otherwise it can be an error for something else. Same for the test below.